specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/availableBalance.json

```diff
@@ -36,66 +79,154 @@
+      "type": "object",
       "allOf": [
         {
           "$ref": "./types.json#/definitions/ArmResource"
         }
       ],
-      "description": "The latest Azure credit balance. This is the balance available for pay now.",
       "properties": {
         "properties": {
-          "$ref": "#/definitions/AvailableBalanceProperties",
-          "title": "Available balance properties",
+          "description": "The properties of a(n) AvailableBalance",
+          "allOf": [
+            {
+              "$ref": "#/definitions/AvailableBalanceProperties"
+            }
+          ],
           "x-ms-client-flatten": true
         }
       },
-      "type": "object"
+      "x-ms-azure-resource": true
     },
     "AvailableBalanceProperties": {
-      "description": "The properties of available balance.",
+      "description": "The available credit or payment on account balance. The credit balance can be used to settle due or past due invoices.",
+      "type": "object",
+      "properties": {
+        "creditAmountForImmediatePayment": {
+          "description": "The amount.",
+          "allOf": [
+            {
+              "$ref": "./types.json#/definitions/Amount"
+            }
+          ]
+        },
+        "paymentsOnAccount": {
+          "description": "The list of payments on accounts.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/PaymentOnAccount"
+          },
+          "readOnly": true,
+          "x-ms-identifiers": []
+        },
+        "totalPaymentsOnAccount": {
+          "description": "The amount.",
+          "allOf": [
+            {
+              "$ref": "./types.json#/definitions/Amount"
+            }
+          ]
+        }
+      }
+    },
+    "PaymentOnAccount": {
+      "description": "An payments on account.",
+      "type": "object",
       "properties": {
         "amount": {
-          "$ref": "./types.json#/definitions/Amount",
-          "description": "Balance amount.",
+          "description": "The amount.",
+          "allOf": [
+            {
+              "$ref": "./types.json#/definitions/Amount"
+            }
+          ]
+        },
+        "billingProfileId": {
+          "description": "The ID of the billing profile for the payments on account.",
+          "type": "string",
+          "readOnly": true
+        },
+        "billingProfileDisplayName": {
+          "description": "The name of the billing profile for the payments on account.",
+          "type": "string",
           "readOnly": true
+        },
+        "invoiceId": {
+          "description": "The ID of the invoice for which the payments on account was generated.",
+          "type": "string",
+          "readOnly": true
+        },
+        "invoiceName": {
+          "description": "The name of the invoice for the payments on account.",
```

Comment:
Can we just have clients grab this from invoiceId?

Reply:
(Keep as-is if there are sufficient precedent for this pattern in other ARM resource APIs...)

Reply:
There is sufficient precedent